### PR TITLE
[#11] Add 404 page.

### LINKED ISSUE #11: Create 404 page with redirect
We need a custom 404 page for the site.

I see that for a page not found the url remains the same. ex https://twisted.org/tra/la

This should be great at it can allow us to do some javascript redirection

@ioanacristinamarinescu can you please take a look at implementing a custom 404 page.

It can be as simple as it gets. The content should be something like this.

Static: Page not found on this site. 
And extra dynamic: We are looking for a redirection to ... URL where you can manually click.
and extra if JS is disable: You should enable JS for this site to take advantage of automatic redirection

Maybe we can do it with tailwinds3 to prepare for a future dark/light theme.

The 404 will also have a redirect JS file.

------------

As an example of redirection. Some simple rules for a start.
 
* https://twisted.org/trac/ticket/4567 -> https://github.com/twisted/twisted-trac-migration-3/issues/4567
* https://twistedmatrix.com/documents/* -> https://docs.twistedmatrix.com/en/stable/

You can try to use the https://github.com/isaacs/minimatch library to match the redirection.

### PR DESCRIPTION
**Pull request by ioanacristinamarinescu**
Fix #11 

Add a 404 page on which there is a link to the github issue that was searched.

Example:
https://twisted.org/trac/ticket/4567 -> https://github.com/twisted/twisted-trac-migration-3/issues/4567

Reviewers: @adiroiban 

To test
======

With JS enabled and JS disable go to this URL:

https://deploy-preview-14--twistedmatrix.netlify.app/trac/ticket/5678/ goes to GitHub ticket page
https://deploy-preview-14--twistedmatrix.netlify.app/tra/la renders 404 page

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -5,10 +5,12 @@
     <meta charset="utf-8">
     <title>Twisted</title>
     <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <link rel="icon" type="image/x-icon" href="assets/images/favicon.ico">
     <link href="assets/css/bootstrap-v4.5.3.min.css" rel="stylesheet" type="text/css" media="all" />
     <link href="assets/css/bootstrap-icons.css" rel="stylesheet" type="text/css" media="all" />
     <link href="assets/css/default.min.css" rel="stylesheet" type="text/css" media="all" />
     <link href="assets/css/style.css" rel="stylesheet" type="text/css" media="all" />
+    <link href="assets/css/tailwind.css" rel="stylesheet" type="text/css" media="all" />
```
Comment (adiroiban) [inline index.html:13]: do we need tailwind.css for the index page?

File: 404.html [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        <a href="#" id="js-link" class="tw-text-highlight">issue</a>
+      </h2>
+      <h2 id="js-enable-js" class="tw-font-medium">
+        Please enable JavaScript for this site to take advantage of automatic redirection.
+      </h2>
+    </div>
+
+    <!-- Required vendor scripts (Do not remove) -->
+    <script type="text/javascript" src="assets/js/jquery-v3.5.1.min.js"></script>
+    <script type="text/javascript" src="assets/js/bootstrap-v4.5.3.min.js"></script>
```
Comment (adiroiban) [inline 404.html:30]: do we need bootstrap here?

File: 404.html [hunk 3]
```diff
@@ -0,0 +1,34 @@
+<!doctype html>
+<html lang="en">
+
+  <head>
+    <meta charset="utf-8">
+    <title>404 - Twisted</title>
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <link rel="icon" type="image/x-icon" href="assets/images/favicon.ico">
+    <link href="assets/css/style.css" rel="stylesheet" type="text/css" media="all" />
```
Comment (adiroiban) [inline 404.html:9]: do we also need the styles.css ?

File: 404.html (hunk 2 above)
Comment (adiroiban) [inline 404.html:30]:
```suggestion
```

File: assets/css/tailwind.css [hunk 4]
```diff
@@ -0,0 +1,214 @@
+*, ::before, ::after {
```
[the remark this answers is not in the document]
Comment (ioanacristinamarinescu) [reply assets/css/tailwind.css:1]: yes, it is generated by tailwindcss.
used `npm run dev` and and the file is updated with the tailwind classes that I used.

File: 404.html (hunk 2 above)
Comment (ioanacristinamarinescu) [reply 404.html]: no

File: 404.html (hunk 3 above)
Comment (ioanacristinamarinescu) [reply 404.html]: no

File: index.html (hunk 1 above)
Comment (ioanacristinamarinescu) [reply index.html]: no

File: assets/css/tailwind.css (hunk 4 above)
[the remark this answers is not in the document]
Comment (adiroiban) [reply assets/css/tailwind.css:1]: Thanks. This needs a readme...and we might want to have a 3-rd directory

* /assets/ - our manual managed files
* /build/ - our files generated by the toosl
* /vender - exact copies for external files

File: tailwind.config.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
+/** @type {import('tailwindcss').Config} */
+module.exports = {
+  content: ['*.html'],
+  prefix: 'tw-',
```
Comment (adiroiban) [inline tailwind.config.js:4]: I like the `tw-` prefix :) very helpful to differentiate custom classes from standard tw classes

**Review by adiroiban — APPROVED** [review]
I think that this is a very good start.

We have the foundation for any future 404 or redirection work.

-----------

Cristina, the changes to the existing readme.md should be as simple as possible. Just to say that the future is tw3, that node/npm is required and what command to run to update the CSS.

Many thanks!

**Comment by ioanacristinamarinescu** [comment]
I have reconfigured the structure of folders.
Automatic redirect to ticket on GitHub.
Other unknown links go to 404 page.
@adiroiban is this ok?

**Pull request merged**